src/zenml/cli/pipeline.py

```diff
@@ -138,3 +144,46 @@ def list_pipeline_runs(
             return
 
         cli_utils.print_pipeline_runs_table(pipeline_runs=pipeline_runs)
+
+
+@runs.command("delete")
+@click.argument("run_name_or_id", type=str, required=True)
+def delete_pipeline_run(run_name_or_id: str) -> None:
+    """Delete a pipeline run.
+
+    Args:
+        run_name_or_id: The name or ID of the pipeline run to delete.
+    """
+    cli_utils.print_active_config()
+    confirmation = cli_utils.confirmation(
+        f"Are you sure you want to delete pipeline run " f"`{run_name_or_id}`?"
+    )
+    if not confirmation:
+        cli_utils.declare("Pipeline run deletion canceled.")
+        return
+
+    client = Client()
+    run = client.get_pipeline_run(name_id_or_prefix=run_name_or_id)
+    run_artifacts: List["ArtifactResponseModel"] = []
+    for step_run in client.list_run_steps(pipeline_run_id=run.id):
+        if step_run.status == ExecutionStatus.COMPLETED:
+            run_artifacts.extend(step_run.output_artifacts.values())
+    if run_artifacts:
+        confirmation = cli_utils.confirmation(
+            f"Pipeline run '{run_name_or_id}' has produced "
+            f"{len(run_artifacts)} artifacts. Do you want to delete these "
+            "artifacts as well?"
+        )
+        if confirmation:
+            for artifact in run_artifacts:
+                client.delete_artifact(artifact_id=artifact.id)
```

Comment:
What happens if the deleted run has a cached step in it? Does the cached artifact also get removed?

Reply:
Nope, if you check above, the artifact is only associated with the step `if step_run.status == ExecutionStatus.COMPLETED`, so not for cached steps